api/capi/src/nnstreamer-capi-pipeline.c

```diff
@@ -302,6 +302,10 @@ cb_bus_sync_message (GstBus * bus, GstMessage * message, gpointer user_data)
         GstState old_state, new_state;
 
         gst_message_parse_state_changed (message, &old_state, &new_state, NULL);
+        if (!pipe_h->isPausedAtDestroy &&
+            old_state == GST_STATE_PLAYING &&
```

Comment:
is it better to only check new_state == GST_STATE_PAUSED? 
IMO, old_state == GST_STATE_PLAYING is unnecessary.

Reply:
I agree with. It's applied

Reply:
Or, add new enum ml_pipeline_state_e pipe_state and set this value when this callback is changed.
then,
```
pipe_h->pipe_state = (ml_pipeline_state_e) new_state;
.....
if (pipe_h->state_cb.cb) {
  pipe_h->state_cb.cb (pipe_h->pipe_state, pipe_h->state_cb.user_data);
}

...
in ml_pipeline_destroy()

if (scret != GST_STATE_CHANGE_FAILURE && state == GST_STATE_PLAYING) {
  state_change_requested = TRUE;
...

while (state_change_requested && pipe_h->pipe_state == ML_PIPELINE_STATE_PLAYING) {
......
}
```

Reply:
Managing the pipeline state looks much better. It's applied!